python/ray/air/_internal/usage.py

```diff
@@ -0,0 +1,299 @@
+from typing import (
+    Callable,
+    Optional,
+    Sequence as Typing_Seq,
+    Type,
+    TYPE_CHECKING,
+    Union,
+)
+
+from collections.abc import Sequence
+from dataclasses import dataclass
+import os
+import json
+
+import ray
+from ray._private.usage.usage_lib import TagKey, record_extra_usage_tag
+
+if TYPE_CHECKING:
+    from ray.tune import Trainable
+    from ray.tune.callback import Callback
+    from ray.tune.experiment import Experiment
+    from ray.tune.schedulers import TrialScheduler
+    from ray.tune.search import SearchAlgorithm, Searcher
+    from ray.tune.search import ConcurrencyLimiter
+    from ray.train.trainer import BaseTrainer
+
+    TrainableType = Union[str, Callable, Type[Trainable]]
+    TrainableTypeOrTrainer = Union[TrainableType, BaseTrainer]
+    TrainableTypeOrTrainerOrExperiment = Union[
+        TrainableTypeOrTrainer, "Experiment", Typing_Seq["Experiment"]
+    ]
+
+
+# All possible entrypoint combinations...
+# * Trainer.fit() --> Tuner.fit()
+# * Trainer.fit() --> Tuner.fit() --> remote --> tuner_internal.fit()
+# * Tuner.fit()
+# * Tuner.fit() --> remote --> tuner_internal.fit()
+# * Tuner.restore() --> Tuner.fit() --> tuner_internal.fit()
+# * Tuner.restore() --> remote --> tuner_internal.fit()
+# * tune.run()
+# * tune.run() --> remote
+# * run_experiment()
+# * run_experiment() --> remote
+
+
+# Entrypoint types
+TRAINER_FIT = "Trainer.fit"
+TUNER_RESTORE = "Tuner.restore"
+TUNER_FIT = "Tuner.fit"
+TUNE_RUN = "tune.run"
+TUNE_RUN_EXP = "tune.run_experiment"
+
+
+# A class that holds information about AIR workload scenario (tune v.s. train)
+# and entrypoint. At various places in the air/train/tune code base, we may access
+# this data structure as a global variable.
+# Ray client mode makes things a bit tricky - caller should understand that this
+# data structure cannot be accessed across different processes and certainly not
+# between client and head node. See below for env vars to bridge client and head.
+# Mainly for telemetry purposes but may also be used for formatting air console output.
+@dataclass
+class AIRScenario:
+    # One of TRAINER_FIT, TUNER_RESTORE, TUNER_FIT, TUNE_RUN, and TUNE_RUN_EXP.
+    entrypoint = None
+
+    ray_client_mode = None
+
+    trainable = None
+    more_than_one_trial = None
+
+    # only meaningful with "more than one trial" case.
+    search_alg = None
+    scheduler_alg = None
```

Comment:
Can we add typehints/whitelist/enumerations to keep track of what is supported here?